src/ralph/operations/models.py

```diff
@@ -0,0 +1,145 @@
+# -*- coding: utf-8 -*-
+from dj.choices import Choices
+from django.conf import settings
+from django.db import models
+from django.db.models.signals import post_migrate
+from django.dispatch import receiver
+from django.utils.translation import ugettext_lazy as _
+from mptt.models import MPTTModel, TreeForeignKey
+
+from ralph.assets.models.base import BaseObject
+from ralph.lib.mixins.fields import NullableCharField
+from ralph.lib.mixins.models import (
+    AdminAbsoluteUrlMixin,
+    NamedMixin,
+    TaggableMixin
+)
+
+
+class OperationStatus(Choices):
+    _ = Choices.Choice
+
+    opened = _('open')
+    in_progress = _('in progress')
+    resolved = _('resolved')
+    closed = _('closed')
+
+
+class OperationType(MPTTModel, NamedMixin, models.Model):
+    parent = TreeForeignKey(
+        'self',
+        null=True,
+        blank=True,
+        related_name='children',
+        db_index=True
+    )
+
+    class choices(Choices):
+        _ = Choices.Choice
+
+        CHANGES = Choices.Group(0)
+        change = _('change')
+
+        INCIDENTS = Choices.Group(100)
+        incident = _('incident')
+
+        PROBLEMS = Choices.Group(200)
+        problem = _('problem')
+
+        FAILURES = Choices.Group(300)
+        failure = _('failure')
+        hardware_failure = _('hardware failure')
+
+
+class Operation(AdminAbsoluteUrlMixin, TaggableMixin, models.Model):
+    type = TreeForeignKey(OperationType, verbose_name=_('type'))
+    title = models.CharField(
+        max_length=350, null=False, blank=False, verbose_name=_('title'),
+    )
+    description = models.TextField(
+        verbose_name=_('description'), null=True, blank=True,
+    )
+    status = models.PositiveIntegerField(
+        verbose_name=_('status'), choices=OperationStatus(),
+    )
+    asignee = models.ForeignKey(
+        settings.AUTH_USER_MODEL,
+        related_name='operations',
+        verbose_name=_('asignee'),
+        null=True,
+        blank=True,
```

Comment:
When `assignee` is deleted `operation` will be also removed, true?
I'm not sure if this is wanted.. . Or perhaps we don't allow removing users.